src/components/kpiWidget/kpiWidget.js

```diff
@@ -1,84 +1,53 @@
 // Copyright (c) Microsoft. All rights reserved.
 
 import React, { Component } from 'react';
+import Rx from 'rxjs';
 import { connect } from 'react-redux';
 import { refreshAllChartData } from '../../actions/kpiActions';
 import PieChartGraph from './pieChartGraph';
 import BarChart from './barChart';
 import { Grid, Row, Col } from 'react-bootstrap';
-import Select from 'react-select';
 import Config from '../../common/config';
 import Lang from '../../common/lang';
 import DashboardPanel from '../dashboardPanel/dashboardPanel';
-import { getNonFunctionalProps } from '../../common/utils';
-import _ from 'lodash';
 
 import './kpiWidget.css';
 
 class KpiWidget extends Component {
-  constructor() {
-    super();
+  constructor(props) {
+    super(props);
+
     this.state = {
-      selectedValue: 'H'
+      timeRange: this.props.timeRange || 'PT1H'
     };
 
-    this.updateValue = this.updateValue.bind(this);
-
-    this.options = [
-      {
-        value: 'H',
-        label: Lang.LASTHOUR
-      },
-      {
-        value: 'D',
-        label: Lang.LASTDAY
-      },
-      {
-        value: 'W',
-        label: Lang.LASTWEEK
-      },
-      {
-        value: 'M',
-        label: Lang.LASTMONTH
-      }
-    ];
-  }
-
-  shouldComponentUpdate(nextProps, nextState) {
-    let nonFunctionalNextProps = _.omit(getNonFunctionalProps(nextProps), ['chartDataFetchComplete']);
-    let nonFunctionalThisProps = _.omit(getNonFunctionalProps(this.props), ['chartDataFetchComplete']);
-    return !_.isEqual(nonFunctionalNextProps, nonFunctionalThisProps) || !_.isEqual(this.state, nextState);
+    this.subscriptions = [];
+    this.errorSubject = new Rx.Subject();
   }
 
-  scheduleAutoUpdate() {
-    this.intervalId = setTimeout(() => {
-      this.props.intervalChanged(this.state.selectedValue);
-    }, Config.INTERVALS.TELEMETRY_UPDATE_MS);
+  componentDidMount() {
+    this.subscriptions.push(
+      Rx.Observable.interval(Config.INTERVALS.TELEMETRY_UPDATE_MS)
+        .startWith(-1)
+        .takeUntil(this.errorSubject)
+        .subscribe(() => this.props.intervalChanged(this.state.timeRange))
+    );
   }
 
   componentWillReceiveProps(nextProps) {
-    if (nextProps.chartDataFetchComplete) {
-      this.scheduleAutoUpdate();
+    const { timeRange } = nextProps;
+    if (timeRange && timeRange !== this.props.timeRange) {
+      this.setState(
+        { timeRange },
+        this.props.intervalChanged(timeRange)
+      )
     }
   }
 
   componentWillUnmount() {
-    if (this.intervalId) {
-      clearInterval(this.intervalId);
-    }
-  }
-
-  componentWillUpdate(nextProps) {
-    if (nextProps.devices) {
-      this.props.intervalChanged(this.state.selectedValue);
-    }
-  }
-
-  updateValue(newValue) {
-    this.setState({
-      selectedValue: newValue
-    });
-    this.props.intervalChanged(newValue);
+    this.subscriptions.forEach(sub => sub.unsubscribe());
+    if (this.getDataSub && this.getDataSub.unsubscribe)
```

Comment:
I don't see getDataSub anywhere. This looks like copy-paste code that doesn't do anything.